common/src/test/java/com/github/dominik48n/party/server/SwitchServerTest.java

```diff
@@ -0,0 +1,214 @@
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.never;
+
+@ExtendWith(MockitoExtension.class)
+class SwitchServerTest {
+
+    private SwitchServer<String> switchServer;
+
+    @Mock
+    private UserManager<String> userManager;
+
+    @Mock
+    private SwitchServerConfig switchServerConfig;
+
+    @Mock
+    private PartyPlayer partyPlayer;
+
+    @Mock
+    private PartyProvider partyProvider;
+
+    @BeforeEach
+    void setUp() throws IllegalAccessException {
+        setPartyProvider();
+        switchServer = createSwitchServer();
+    }
+
+    @Test
+    void handleServerConnected() throws JsonProcessingException {
+        String user = "USER";
+        String serverName = "Lobby3";
+        UUID leaderUUID = UUID.randomUUID();
+        UUID partyUuid = UUID.randomUUID();
+        Party party = createParty(partyUuid, leaderUUID);
+
+        when(userManager.getPlayer(user)).thenReturn(Optional.of(partyPlayer));
+        when(partyPlayer.partyId()).thenReturn(Optional.of(partyUuid));
+        when(partyPlayer.uniqueId()).thenReturn(leaderUUID);
+        when(partyProvider.getParty(partyUuid)).thenReturn(Optional.of(party));
+
+        when(switchServerConfig.blackEnable()).thenReturn(true);
+        when(switchServerConfig.blackPatternList()).thenReturn(of(Pattern.compile("^Lobby.*")));
+        when(switchServerConfig.whiteEnable()).thenReturn(true);
+        when(switchServerConfig.whitePatternList()).thenReturn(of(Pattern.compile("Lobby3")));
+
+        switchServer.handleServerConnected(user, serverName);
+
+        verify(partyProvider).connectPartyToServer(party, serverName);
+    }
+
+    @Test
+    void handleServerConnectedFailBecauseNotMatchWhiteList() throws JsonProcessingException {
+        String user = "USER";
+        String serverName = "Lobby2";
+        UUID leaderUUID = UUID.randomUUID();
+        UUID partyUuid = UUID.randomUUID();
+        Party party = createParty(partyUuid, leaderUUID);
+
+        when(userManager.getPlayer(user)).thenReturn(Optional.of(partyPlayer));
+        when(partyPlayer.partyId()).thenReturn(Optional.of(partyUuid));
+        when(partyPlayer.uniqueId()).thenReturn(leaderUUID);
+        when(partyProvider.getParty(partyUuid)).thenReturn(Optional.of(party));
+
+        when(switchServerConfig.blackEnable()).thenReturn(true);
+        when(switchServerConfig.blackPatternList()).thenReturn(of(Pattern.compile("^Lobby.*")));
+        when(switchServerConfig.whiteEnable()).thenReturn(true);
+        when(switchServerConfig.whitePatternList()).thenReturn(of(Pattern.compile("Lobby3")));
+
+        switchServer.handleServerConnected(user, serverName);
+
+        verify(partyProvider, never()).connectPartyToServer(any(), any());
+    }
+
+    @Test
+    void handleServerConnectedDisableWhiteList() throws JsonProcessingException {
```

Comment:
Please declare the variables in the method that should not be changed as `final`.